modules/sfm/CMakeLists.txt

```diff
@@ -0,0 +1,83 @@
+set(the_description "SFM algorithms")
+
+ocv_warnings_disable(CMAKE_CXX_FLAGS -Wundef -Winvalid-pch)
+
+# needed since we are linking static libs
+set(CMAKE_CXX_FLAGS ${CMAKE_CSS_FLAGS} "-fPIC")
+
+
+### LIBMV LIGHT DEFINITIONS ###
+
+set(LIBMV_LIGHT_INCLUDES
+  src/libmv_light
+  src/libmv_light/third_party/ssba
+)
+
+set(LIBMV_LIGHT_LIBS
+  reconstruction
+  correspondence
+  camera
+  multiview
+  multiview_test_data
+  numeric
+)
+
+
+### LIBMV LIGHT EXTERNAL DEPENDENCIES ###
+
+find_package( Ceres )
+
+if ( Ceres_FOUND )
+  add_definitions("-DCERES_FOUND=1")
+  list(APPEND LIBMV_LIGHT_LIBS simple_pipeline)
+else ()
+  add_definitions("-DCERES_FOUND=0")
+endif ()
+
+
+### CHECK OPENCV SFM MODULE DEPENDENCIES ###
+
+set(OPENCV_SFM_DEPS
+  opencv_core
+  opencv_ts
+  opencv_calib3d
+  opencv_features2d
+  opencv_xfeatures2d
+  opencv_viz
+)
+
+ocv_check_dependencies(${OPENCV_SFM_DEPS})
+ocv_include_directories("${OpenCV_SOURCE_DIR}/include/opencv")
+ocv_include_modules_recurse(${OPENCV_SFM_DEPS})
```

Comment:
Module's CMakeLists.txt files are processed twice:
- 1st pass: collect dependency information. Only lines before `ocv_add_module` are processed.
- 2nd pass (only if module dependencies are available and module is not disabled): configure targets to build.

There is no correct info during 1st pass about all modules so `ocv_check_dependencies` and `ocv_include_modules_recurse` doesn't work properly.

My suggestion is to put these lines near `add_subdirectory(src/libmv_light)`. Lines after `ocv_add_module` executed only during 2nd pass when modules information is available.